src/cargo/ops/cargo_package.rs

```diff
@@ -523,18 +523,15 @@ fn check_repo_state(
                     }));
                 }
             }
-            config.shell().verbose(|shell| {
-                shell.warn(format!(
-                    "No (git) Cargo.toml found at `{}` in workdir `{}`",
-                    path.display(),
-                    workdir.display()
-                ))
-            })?;
+            config.emit_verbose_diagnostic(format!(
+                "No (git) Cargo.toml found at `{}` in workdir `{}`",
+                path.display(),
+                workdir.display()
+            ))?;
         }
     } else {
-        config.shell().verbose(|shell| {
-            shell.warn(format!("No (git) VCS found for `{}`", p.root().display()))
-        })?;
+        config
+            .emit_verbose_diagnostic(format!("No (git) VCS found for `{}`", p.root().display()))?;
```

Comment:
I believe these are the only `emit_verbose_diagnostic`.  

How should these be handled?
- In the future, put these under a `pedantic` group and remove `emit_verbose_diagnostic`
- Always display these
- Not make these diagnostics but more like logging messages